src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -200,10 +204,14 @@ public function has($id)
             if ('service_container' === $id
                 || isset($this->aliases[$id])
                 || isset($this->services[$id])
-                || array_key_exists($id, $this->services)
             ) {
                 return true;
             }
+            if (array_key_exists($id, $this->services)) {
+                @trigger_error(sprintf('Checking for the existence of a unset "%s" service is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0.', $id), E_USER_DEPRECATED);
```

Comment:
this does not make sense, as there won't be any unset services in 4.0, and code calling ``has()`` has no way to know whether the service instance was reset or no.

And btw, you change the code to unset the array entry instead of setting it to ``null``, so this ``if`` won't match anymore